jina/executors/indexers/cache.py

```diff
@@ -92,39 +90,34 @@ def query(self, data, *args, **kwargs) -> Optional[bool]:
         if self.query_handler is None:
             self.query_handler = self.get_query_handler()
 
-        if self.field == ID_KEY:
-            status = (data in self.query_handler.ids) or None
-        else:
-            status = (data in self.query_handler.content_hash) or None
+        return data in self.query_handler.cache_val_to_id
 
-        return status
 
-    def update(self, keys: Iterable['UniqueId'], values: Iterable[any], *args, **kwargs):
+    def update(self, keys: Iterable[str], values: Iterable[any], *args, **kwargs):
         """Update cached documents.
         :param keys: list of Document.id
         :param values: list of either `id` or `content_hash` of :class:`Document`"""
         # if we don't cache anything else, no need
```

Comment:
here

Reply:
That one is still valid.
If we only cache the id, the update should not do anything.
The content is getting updated but the id will stay the same.